presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MVRegistry.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.optimizations;
+
+import com.facebook.presto.metadata.QualifiedObjectName;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MVRegistry
+{
+    private final Map<QualifiedObjectName, List<MVInfo>> registry;
+
+    public MVRegistry()
+    {
+        this.registry = new HashMap<>();
+        init();
+    }
+
+    public void init()
+    {
+        addSimpleMVTable();
+        addAdMetricsTable();
+    }
+
+    private void addAdMetricsTable()
```

Comment:
We should be able to infer this in the statement analyzer by examining the AST. Will add this point to design doc.